plugin.js

```diff
@@ -58,45 +58,79 @@ const add = async function (context) {
-        insert: `apply plugin: 'com.google.gms.google-services'`,
-    });
-    ignite.patchInFile(`${process.cwd()}/ios/${name}/AppDelegate.m`, {
-        before: '@implementation AppDelegate',
-        insert: `@import Firebase;
-`,
-    });
-    ignite.patchInFile(`${process.cwd()}/ios/${name}/AppDelegate.m`, {
-        after: 'NSURL \\*jsCodeLocation;',
-        insert: `  [FIRApp configure];`,
-    });
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/build.gradle`, {
+            after: 'dependencies {',
+            insert: `        classpath 'io.fabric.tools:gradle:1.25.4'`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/build.gradle`, {
+            after: 'dependencies {',
+            insert: `        classpath 'com.google.gms:google-services:4.1.0'`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/build.gradle`, {
+            after: 'mavenLocal()',
+            insert: `        google()`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/app/build.gradle`, {
+            after: 'apply plugin: "com.android.application"',
+            insert: `apply plugin: "io.fabric"`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/app/build.gradle`, {
+            after: 'implementation "com\\.facebook\\.react:react-native',
+            insert: `    implementation "com.crashlytics.sdk.android:crashlytics:2.9.5"`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/app/build.gradle`, {
+            after: 'compile "com\\.facebook\\.react:react-native',
+            insert: `    compile "com.google.firebase:firebase-core:16.0.4"`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/android/app/build.gradle`, {
+            before: '// Run this once to be able to run the application with BUCK',
+            insert: `apply plugin: 'com.google.gms.google-services'`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/ios/${name}/AppDelegate.m`, {
+            before: '@implementation AppDelegate',
+            insert: `@import Firebase;
+    `,
+        });
+    } catch (error) {
+        console.error(error);
+    }
+    try {
+        ignite.patchInFile(`${process.cwd()}/ios/${name}/AppDelegate.m`, {
+            after: 'NSURL \\*jsCodeLocation;',
+            insert: `  [FIRApp configure];`,
+        });
+    } catch (error) {
+        console.error(error);
+    }
```

Comment:
not sure how I feel about wrapping each of these in try catches. It makes sense to prevent the halt of execution if there's something missing, but at the same time dont we need all these strings to be replaced for the installation to be successful?

Reply:
Problem is, right now we don't SEE any of these errors. So I feel it's better for it to fail one step, you see it and you have to fix it manually than not knowing where it failed and why

Reply:
Makes sense!